### PR TITLE
Log error when statsd server fails to start and fix airflow docs

### PR DESCRIPTION
**Pull request by belimawr**
## Proposed commit message
This commit adds a log error when the statsd server fails to start, it
also improves the Airflow module documentation to make extra clear
that the module will start a statsd server.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

~~## Author's Checklist~~
## How to test this PR locally
1. Create the following `metricbeat.yml`:
    ```yaml
    metricbeat.modules:
    - module: airflow
      host: "1.2.3.4" # Any IP that does not belong to the host running Metricbeat
      port: "9102"
      ttl: "10s"
      period: 5s
      metricsets: [ 'statsd' ]
    
    output.console:
      enabled: true
    ```
2. Start Metricbeat
3. After a few seconds you should see a log error like this:
    ```json
    {"log.level":"error","@timestamp":"2023-08-31T17:09:34.565+0200","log.logger":"airflow.statsd","log.origin":{"file.name":"server/server.go","file.line":201},"message":"could not start statsd server for module: airflow","service.name":"metricbeat","error":{"message":"failed to start UDP server: listen udp 1.2.3.4:9102: bind: cannot assign requested address"},"ecs.version":"1.6.0"}
    ```
~~## Related issues~~
~~## Use cases~~
~~## Screenshots~~
~~## Logs~~

### CODE REVIEW

**Review by AndersonQ — APPROVED** [review]

**Comment by shmsr** [comment]
Looks good to me but I could see more adjustments that could be made to Airflow docs. Sample patch:

```diff
 This module collects metrics from
-https://airflow.apache.org/docs/apache-airflow/stable/logging-monitoring/metrics.html[Airflow metrics]. It runs a
-statsd server where airflow will send metrics to. The default metricset is `statsd`.
+https://airflow.apache.org/docs/apache-airflow/stable/logging-monitoring/metrics.html[Airflow metrics]. It runs a statsd server where airflow will send metrics to. The default metricset is `statsd`.
 
 [float]
 === Compatibility
 
-The Airflow module is tested with Airflow 2.1.0. It should work with version
-2.0.0 and later.
+The Airflow module is tested with Airflow 2.1.0. It should work with version 2.0.0 and later.
 
 [float]
 === Usage
-The Airflow module requires <<metricbeat-module-statsd,Statsd>> to receive Statsd metrics. Refer to the link for instructions about how to use Statsd.
+The Airflow module requires <<metricbeat-module-statsd,Statsd>> to receive statsd metrics. Refer to the link for instructions about how to use statsd.
 
-Add the following lines to your Airflow configuration file e.g. `airflow.cfg` ensuring `statsd_prefix` is left empty and replace `%METRICBEAT_HOST%` with the address metricbeat is running:
+Add the following lines to your Airflow configuration file e.g. `airflow.cfg` ensuring `statsd_prefix` is left empty and replace `%METRICBEAT_HOST%` with the address where metricbeat is running:
```

**Review by shmsr — CHANGES_REQUESTED** [review]
Left one comment.

**Comment by belimawr** [comment]
@shmsr I added all your suggestions on e8dc1d693c02d3eb72bef7456c7b9f340472b420

**Review by shmsr — APPROVED** [review]

File: x-pack/metricbeat/module/statsd/server/server.go
Comment (shmsr) [inline x-pack/metricbeat/module/statsd/server/server.go:1]: nitpick: rename the method receiver from `b` to `m`: s/b/m

Comment (shmsr) [reply x-pack/metricbeat/module/statsd/server/server.go:1]: not important though

Comment (belimawr) [reply x-pack/metricbeat/module/statsd/server/server.go:1]: That's a bigger change on the this file, I rather make it into a separated PR.

Comment (belimawr) [reply x-pack/metricbeat/module/statsd/server/server.go:1]: The followup PR: https://github.com/elastic/beats/pull/36591

I think when I first looked I was thinking about the opposite change :man_facepalming:. It's a super small PR :D

**Pull request merged**